8422: Login And Register Endpoints

HBA114 opened the pull request:
Created required constants, models and controllers

Commit:
Fake OTP and Authorization Endpoints

Created required constants, models and controllers

Commit:
Updated BaseController and Fixed Misspelled Naming on SendOtpController

Changed file name and class name for SendOtpController,
Added summaries to BaseController

Commit:
Created Entities and removed unnecessary package from Domain Layer

Commit:
Created DbContext, EntityConfigurations And DependencyInjection

Updated package versions, added Microsoft.AspNetCore.Identity package. Created InfrastructureServiceRegistration dependency injection file

Commit:
Added Required Packages To API, Registered Infrastructure Services

Commit:
Updated Station and County Entities, Updated Configurations

Removed Country reference from County, removed Country and City reference from Station. Removed configurations related to removed entity references

kesoyd commented on Plugzy.API/Controllers/BaseController.cs:
```diff
@@ -0,0 +1,32 @@
+    /// <Summary>
+    ///     Base Controller with Mediator
+    /// </Summary>
+    protected IMediator Mediator => _mediator ??= HttpContext.RequestServices.GetService<IMediator>()!;
```
base controller constructor ı direk DI ile alabileceğin şekilde revize edebilirsin,  her seferinde  servisleri çağırmana gerek yok.

genel olarak naming için  constructorda kullandığın ve erişilebilecek obkeler için  _mediator  gibi kullan.  yani nihai kullanılan obje adı  _xxxYyyy  gibi olsun

HBA114 replied:
Commit ile çözüldü.
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/43145fe4f33ace05db8e5db63a42bc62ef88fc52

kesoyd commented on Plugzy.API/Controllers/SendOtp.cs:
```diff
@@ -0,0 +1,22 @@
+///<Summary>
+/// OTP Controller
+///</Summary>
+public class SendOtp : BaseController
```
her işlem için ayrı controller a gerek yok,   işlemleri gruplayarak controller oluşturabilirsin.  controller ismlerinde Controller olsun mutlaka.

bu işlemi authorizecontroller a atabilirsin mesela

HBA114 replied:
Controller isimlerini hemen düzeltiyorum.
Ayrı controller olarak oluşturma sebebim görsellerdeki url uzantıları aslında. Bir klasör olarak mı gruplayayım yoksa bu iki işlemi tek bir controller altında mı gerçekleştireyim?
![post_auth](https://github.com/gkhntpc/PlugzyApp/assets/55455410/fb66e371-b372-4d0a-a07f-b6a40581df46)
![post_otp](https://github.com/gkhntpc/PlugzyApp/assets/55455410/795668ac-dcf5-4a03-bc8d-5adc742d939f)

HBA114 replied:
Commit ile çözüldü.
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/43145fe4f33ace05db8e5db63a42bc62ef88fc52

kesoyd commented on Plugzy.Service/Features/Authorization/Commands/AuthorizeCommand.cs:
```diff
@@ -0,0 +1,50 @@
+
+            // validate otp, if true continue else fail result
+
+            string fakeOtp = "1234";
```
bu değerler appsettings den gelsin

HBA114 replied:
Commit ile çözüldü.
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/c5c84df823167f5550b889cfdd1a0e1d03d1746f

kesoyd commented on Plugzy.Utilities/Constants/Authorization/AuthorizationType.cs:
```diff
@@ -0,0 +1,7 @@
+
+public static class AuthorizationType 
+{
+    public const int Login = 1;
```
bunları  static class yerine enum olarak yapabilirsin

HBA114 replied:
Commit ile çözüldü.
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/c5c84df823167f5550b889cfdd1a0e1d03d1746f

Commit:
Changed Base Controller Service Registration, Organised Controllers

Renamed AuthorizeController as AuthorizationController, removed SendOtp(Controller), created "sendOpt" endpoint in AuthorizationController.

Commit:
Added configuration for FakeAuth, used configuration in AuthorizeCommand

Added FakeAuth configurations, used configurations in AuthorizeCommand, changed static class AuthorizationType to enum.

Commit:
Changed Endpoint Return Types, Updated OtpResponse and RequestOtpCommand

Commit:
Removed Entity Configurations, Configured with Attributes

Removed all entity configurations, configured entities using attributes. Added required connection string.

Commit:
Updated DbContext options

Commit:
Merge branch '8422' into entities_and_database_context

Commit:
Merge pull request #1 from HBA114/entities_and_database_context

Entities And DbContext Created

Commit:
Created Jwt Service, updated AuthorizeCommand

Created TokenOptions and ProviderName in appsettings.json.
Created JwtService with AccessToken and RefreshToken generation and registered with AddInfrastructureServices.
Added TokenProvider to Identity service. Created UserType enum.
Updated AuthorizeCommand as using database and creating tokens.

kesoyd commented on Plugzy.API/Plugzy.API.csproj:
```diff
@@ -6,18 +6,24 @@
 
   <ItemGroup>
     <ProjectReference Include="..\Plugzy.Service\Plugzy.Service.csproj" />
+    <ProjectReference Include="..\Plugzy.Infrastructure\Plugzy.Infrastructure.csproj" />
```
service zaten infra ya depend ediyor,  neden ayrıca ekledin?  aşağı doğru her katmana eklemişsin sanki

HBA114 replied:
Commit ile çözüldü
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/d251202ccc9cc1b42ae55625244c29870f791fb6

kesoyd commented on Plugzy.API/Plugzy.API.csproj:
```diff
@@ -6,18 +6,24 @@
   <ItemGroup>
     <PackageReference Include="AspNetCore.HealthChecks.UI.Client" Version="6.0.5" />
     <PackageReference Include="FluentValidation" Version="11.6.0" />
+    <PackageReference Include="MediatR" Version="11.1.0" />
```
model içinde var zaten burada ek olarak gerek yok.  package larda  versiyon uyuşmazlığı ve gereksiz kullanım yapmamaya çalış

HBA114 replied:
Commit ile çözüldü
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/d251202ccc9cc1b42ae55625244c29870f791fb6

kesoyd commented on Plugzy.Service/Features/Authorization/Commands/AuthorizeCommand.cs:
```diff
@@ -0,0 +1,90 @@
+            // Fail if OTP Code is invalid
+            if (request.AuthorizationRequest.otpCode != fakeOtp)
+            {
+                return CommandResult<AuthorizationResponse>.GetFailed("Hatalı OTP Kodu!");
```
tüm b utarz string message ları  tek bir "static MessageCOnstants"   altında toplayıp oradan yönetir misin

HBA114 replied:
Commit ile çözüldü
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/d251202ccc9cc1b42ae55625244c29870f791fb6

kesoyd commented on Plugzy.Service/Features/Otp/Commands/RequestOtpCommand.cs:
```diff
@@ -0,0 +1,41 @@
+
+            response.Seconds = fakeSeconds;
+            response.Confirm = fakeConfirm;
+            response.Code = int.Parse(_configuration["FakeAuth:Otp"]!);
```
artık response da döndüğüne göre fake otp kullanmana gerek yok üretip onu dönebilir ve diğer metotta valide edebilirsin

HBA114 replied:
Commit ile çözüldü
https://github.com/gkhntpc/PlugzyApp/pull/14/commits/434316629196739dbb79692171492665811f0c9c

Commit:
Removed unnecessary references, Created Message Constant

Created AuthorizationMessageConstants and used in AuthorizeCommand, removed Infrastructure reference from API, registered Infrastructure services in Services(Application) layer.

Commit:
Created Generic Repository, created Otp Repository, Otp Logic Completed

Created Otp Message constants, added&updated messages on Authorization Message constants. Created generic repository with CRUD methods, created OtpRepository. Implemented OTP code logic to RequestOtpCommand. Used created otp command in AuthorizeCommand. Register and Login is completed for Mobile.
